### PR TITLE
Allow new window as administrator

### LINKED ISSUE #259: Add "New Root Window" option
Just like in Terminal, can you make such an option in the context menu? So, I right-click Scratch icon and select "New Root Window" and then I can open and edit files as root.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/51155854-add-new-root-window-option?utm_campaign=plugin&utm_content=tracker%2F61917289&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F61917289&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mdh34**
Adds pkexec support to allow the user to open a window as root - fixes #259 
Note: most of this code is derived from files' admin implementation

### CODE REVIEW

File: data/io.elementary.code.policy.xml.in.in [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE policyconfig PUBLIC
+ "-//freedesktop//DTD PolicyKit Policy Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/PolicyKit/1.0/policyconfig.dtd">
+<policyconfig>
+  <vendor>elementary</vendor>
+  <vendor_url>https://github.com/elementary/scratch</vendor_url>
+  <icon_name>system-file-manager</icon_name>
```
Comment (jeremypw) [inline data/io.elementary.code.policy.xml.in.in]: This needs adjusting to the correct icon

**Comment by jeremypw** [comment]
Apologies - this has to be blocked pending resolution of debate concerning the "Use as Administrator" option in elementaryos in general.  It may be removed from Files in future.

**Comment by jlnr** [comment]
Cross-linking the Files issue: https://github.com/elementary/files/issues/224

Just as in Files, I think a better design would be to open an auth prompt when a user tries to save a file that they don't have write permissions for. I don't know how this would work with Code's auto-save model though.

**Comment by mdh34** [comment]
closing this due to UX flaws, conflicts and the new admin:// backend

**Pull request closed without merging**